tonic/datasets/cached_dataset.py

```diff
@@ -0,0 +1,90 @@
+import os
+from typing import Callable, Optional, Tuple, Iterable, Union
+from warnings import warn
+from dataclasses import dataclass
+import h5py
+from pathlib import Path
+import hashlib
+
+
+def save_to_cache(processed_data, fname: Union[str, Path]) -> None:
+    """
+    Save data to cache
+    Args:
+        processed_data:
+        fname:
+    """
+    with h5py.File(fname, "w") as f:
+        if type(processed_data) == tuple:  # can be events, frames, imu, gps, etc.
+            for j, data_piece in enumerate(processed_data):
+                f.create_dataset(f'{j}', data=data_piece)
+        else:
+            f.create_dataset(str(0), data=processed_data)
+
+
+def load_from_cache(fname: Union[str, Path]) -> Tuple:
+    """
+    Load data from cache
+    Args:
+        fname:
+    Returns:
+        data
+    """
+    with h5py.File(fname, "r") as f:
+        data = tuple(f[key][()] for key in f.keys())
+    return data
+
+
+@dataclass
+class CachedDataset:
```

Comment:
this class currently hashes the augmentations, but what it really should be hashing is the preprocessing transforms. With current behaviour, if I change my augmentation, it has to recache.

Reply:
The only place it is hashing is line 89 ie. transforms. Nothing else changes.